source/MaterialXTest/MaterialXRuntime/Runtime.cpp

```diff
@@ -2431,4 +2431,71 @@ TEST_CASE("Runtime: logging", "[runtime]")
     REQUIRE("Info: Test" == logger->result);
 }
 
+TEST_CASE("Runtime: duplicate name", "[runtime]")
+{
+    mx::RtScopedApiHandle api;
+
+    mx::RtStagePtr stage = api->createStage(ROOT);
+
+    // Create a new nodedef for an add node.
+    mx::RtNodeDef addFloat = stage->createPrim("/ND_add_float", mx::RtNodeDef::typeName());
+    addFloat.setNode(ADD);
+    addFloat.createInput(IN1, mx::RtType::FLOAT);
+    addFloat.createInput(IN2, mx::RtType::FLOAT);
+    addFloat.createOutput(OUT, mx::RtType::FLOAT);
+    addFloat.registerMasterPrim();
+
+    // Create a nodegraph object.
+    mx::RtNodeGraph graph1 = stage->createPrim("/graph1", mx::RtNodeGraph::typeName());
+    REQUIRE(graph1.isValid());
+
+    // Create add node in the graph.
+    mx::RtNode add1Node = stage->createPrim("/graph1/add1", addFloat.getName());
+    REQUIRE(graph1.getNode(add1Node.getName()));
+
+    // Add an interface to the graph.
+    mx::RtInput Ainput = graph1.createInput(A, mx::RtType::FLOAT);
+    graph1.createOutput(OUT, mx::RtType::FLOAT);
+    REQUIRE(graph1.getPrim().getAttribute(A));
+    REQUIRE(graph1.getPrim().getAttribute(OUT));
+    REQUIRE(graph1.getInput(A));
+    REQUIRE(graph1.getOutput(OUT));
+    REQUIRE(graph1.getInputSocket(A));
+    REQUIRE(graph1.getOutputSocket(OUT));
+
+    mx::RtToken add1("add1");
+    mx::RtToken add2("add2");
+    mx::RtToken add3("add3");
+    mx::RtToken add4("add4");
+    mx::RtToken add5("add5");
+
+    // Test renaming the input to the name of the node
+    graph1.renameInput(A, add1);
+    REQUIRE(!graph1.getInput(A));
+    REQUIRE(!graph1.getInputSocket(A));
+    REQUIRE(!graph1.getInput(add1));
+    REQUIRE(!graph1.getInputSocket(add1));
+    REQUIRE(graph1.getInput(add2));
```

Comment:
Instead of this you can use the return value of renameInput() to avoid any issue if the internal logic for name generation every changes and avoid hard-coding the outcome.

Reply:
I don't think you can do that because what if the return value from renameInput is wrong? You'd still have to compare renameInput's result against the expected result which is essentially the same as what is being done anyways.